### ISSUE TITLE
Solr 'Restore backup' doesn't bring back .data directory

### ISSUE THREAD

**Issue by renekrie** [querqy/querqy#263]
When restoring a backup in Solr, the rewriter data directory (`.data`) will not be restored to ZooKeeper as the dot in the name makes it a hidden directory.

Solution: rename the directory.

**Comment by renekrie** [comment]
Solution:

* Make directory name configurable in solrconfig.xml. `.data` will remain the default.
* Rewriter meta data is stored in a ZK node. The rewriter id is used as the node name. (This is the status quo). The rewriter data (`.data` directory) is located under the same parent node as the rewriter meta data. This means that a rewriter should not have the ID `.data` or whatever the configurable directory name is. We will validate rewriter names not to start with a dot (as it has already been implemented) but also start excluding the configured data directory name from the valid rewriter IDs. Strictly spoken, in ZK, also a directory could have data (everything is just a node), so that it shouldn't be a big problem if someone changes the configuration so that a former rewriter id becomes the name of the data directory.
* We'll extend the rewriter meta data so that it includes the path to the rewriter data.

Node/directory structure in ZK:

```
/configs/<zkConfigName, default=collection name>/querqy/rewriters/<rewriter data dir, default=.data>
```

Let's say we have a rewriter `common_rules` and use the `.data` directory. Its configuration would be so large, that it needs to be split into 2 files. We'd see the following files:

```
/configs/<zkConfigName>/querqy/rewriters/.data/common_rules-uuid1 - config part 1
/configs/<zkConfigName>/querqy/rewriters/.data/common_rules-uuid2 - config part 2
/configs/<zkConfigName>/querqy/rewriters/common_rules - rewriter metadata
```

The rewriter meta data file currently holds the list of config parts (`common_rules-uuid1`,`common_rules-uuid1`). We'll add the data path (`.data` , or whatever was configured in solrconfig.xml).

To facilitate migration, we will still be able to read the old format (without the data path) and assume `.data`.

**Comment by renekrie** [comment]
Fixed by #274
